specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json

```diff
@@ -0,0 +1,2908 @@
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "private link resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "private link resource type"
+        }
+      },
+      "description": "A private link resource"
+    },
+    "PrivateLinkResourceProperties": {
+      "properties": {
+        "groupId": {
+          "description": "The private link resource group id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "requiredMembers": {
+          "description": "The private link resource required member names.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "requiredZoneNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The private link resource DNS zone name."
+        }
+      },
+      "description": "Properties of a private link resource."
+    },
+    "DiskRestorePoint": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DiskRestorePointProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "description": "Properties of disk restore point"
+    },
+    "DiskRestorePointProperties": {
+      "properties": {
+        "timeCreated": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The timestamp of restorePoint creation",
+          "readOnly": true
+        },
+        "sourceResourceId": {
+          "type": "string",
+          "description": "arm id of source disk",
+          "readOnly": true
+        },
+        "osType": {
+          "type": "string",
+          "description": "The Operating System type.",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "x-ms-enum": {
+            "name": "OperatingSystemTypes",
+            "modelAsString": false
+          }
+        },
+        "hyperVGeneration": {
+          "type": "string",
+          "description": "The hypervisor generation of the Virtual Machine. Applicable to OS disks only.",
+          "enum": [
+            "V1",
+            "V2"
+          ],
+          "x-ms-enum": {
+            "name": "HyperVGeneration",
+            "modelAsString": true
+          }
+        },
+        "familyId": {
+          "type": "string",
+          "description": "id of the backing snapshot's MIS family",
+          "readOnly": true
+        },
+        "sourceUniqueId": {
+          "type": "string",
+          "description": "unique incarnation id of the source disk",
+          "readOnly": true
+        },
+        "diskEncryptionSet": {
+          "$ref": "#/definitions/DiskEncryptionSet"
```

Comment:
this should be the resource ID of the disk encryption set, not the full resource.  A user should require read access on the disk encryption set to see its full properties

Reply:
It should also be marked readonly

Reply:
updated to "encryption"